bot/exts/moderation/infraction/management.py

```diff
@@ -45,6 +45,52 @@ async def infraction_group(self, ctx: Context) -> None:
         """Infraction manipulation commands."""
         await ctx.send_help(ctx.command)
 
+    @infraction_group.command(name="append", aliases=("amend", "add", "a"))
+    async def infraction_append(
+        self,
+        ctx: Context,
+        infraction_id: t.Union[int, allowed_strings("l", "last", "recent")],  # noqa: F821
+        duration: t.Union[Expiry, allowed_strings("p", "permanent"), None],   # noqa: F821
+        *,
+        reason: str = None
+    ) -> None:
+        """
+        Append text and/or edit the duration of an infraction.
+
+        Durations are relative to the time of updating and should be appended with a unit of time.
+        Units (∗case-sensitive):
+        \u2003`y` - years
+        \u2003`m` - months∗
+        \u2003`w` - weeks
+        \u2003`d` - days
+        \u2003`h` - hours
+        \u2003`M` - minutes∗
+        \u2003`s` - seconds
+
+        Use "l", "last", or "recent" as the infraction ID to specify that the most recent infraction
+        authored by the command invoker should be edited.
+
+        Use "p" or "permanent" to mark the infraction as permanent. Alternatively, an ISO 8601
+        timestamp can be provided for the duration.
+        """
+        if isinstance(infraction_id, str):
+            old_infraction = await self.get_latest_infraction(ctx.author.id)
+
+            if old_infraction is None:
+                await ctx.send(
+                    ":x: Couldn't find most recent infraction; you have never given an infraction."
+                )
+                return
+
+            infraction_id = old_infraction["id"]
+
+        else:
+            old_infraction = await self.bot.api_client.get(f"bot/infractions/{infraction_id}")
```

Comment:
Since this is all repeated in the other command, it should all be in a separate function. At this point, I think you should make it a converter. It'll allow you to get rid of `allowed_strings("l", "last", "recent")` since you'll be able to internalise that into the converter.

Reply:
Should the converter return the infraction id or the infraction dictionary itself? I'm thinking of something like
```python
class InfractionLookup(Converter):
    async def convert(self, ctx: Context, arg: str) -> t.Optional[dict]:
```
although I'm not sure whether the `infraction_id` parameter reflects its purpose concisely.

Reply:
Return the full infraction data as a dictionary. Otherwise, you'd need to make a redundant request to get the data within the command. This is cause to get the ID, the converter has to make an API request in the case that the user wants to use the most recent infraction.